src/Tmds.Ssh/SshAgent.cs

```diff
@@ -23,61 +23,77 @@ sealed class SshAgent : IDisposable
-                {
-                    return null;
-                }
-
-                if (string.IsNullOrEmpty(sshAuthSock))
-                {
-                    return null;
-                }
-
-                _defaultEndPoint ??= new UnixDomainSocketEndPoint(sshAuthSock);
-
-                return _defaultEndPoint;
+                return OperatingSystem.IsWindows()
+                    ? "openssh-ssh-agent"
+                    : Environment.GetEnvironmentVariable("SSH_AUTH_SOCK");
             }
         }
 
-        private readonly EndPoint _endPoint;
+        private readonly string _address;
         private readonly SequencePool _sequencePool;
 
-        private SocketSshConnection? _agentConnection;
+        private StreamSshConnection? _agentConnection;
 
-        public SshAgent(EndPoint endPoint, SequencePool sequencePool)
+        public SshAgent(string address, SequencePool sequencePool)
         {
-            _endPoint = endPoint;
+            _address = address;
             _sequencePool = sequencePool;
         }
 
         public async ValueTask<bool> TryConnect(CancellationToken cancellationToken)
         {
-            Socket? socket = null;
+            Stream? stream = null;
             try
             {
-                socket = new Socket(AddressFamily.Unix, SocketType.Stream, ProtocolType.Unspecified);
-                await socket.ConnectAsync(_endPoint, cancellationToken).ConfigureAwait(false);
-                _agentConnection = new SocketSshConnection(NullLoggerFactory.Instance.CreateLogger<SshClient>(), _sequencePool, socket);
+                var logger = NullLoggerFactory.Instance.CreateLogger<SshClient>();
+
+                if (OperatingSystem.IsWindows())
+                {
+                    // There is no easy way to detect if a named pipe exists or
+                    // not. File.Exists is the fastest but will actually open a
+                    // connection and close it. For the ssh-agent implementation
+                    // that won't matter as it is configured to allow multiple
+                    // connections.
+                    if (!File.Exists(@$"\\.\pipe\{_address}"))
+                    {
+                        return false;
+                    }
+
+                    NamedPipeClientStream pipe = new NamedPipeClientStream(
+                        ".",
+                        _address,
+                        PipeDirection.InOut,
+                        PipeOptions.Asynchronous,
+                        System.Security.Principal.TokenImpersonationLevel.Anonymous);
+                    stream = pipe;
+
+                    await pipe.ConnectAsync(1000, cancellationToken).ConfigureAwait(false);
+                }
+                else
+                {
+                    Socket socket = new Socket(AddressFamily.Unix, SocketType.Stream, ProtocolType.Unspecified);
+                    stream = new NetworkStream(socket);
```

Comment:
The default NetworkStream ctor doesn't own the Socket, you need to call the overload that accepts a boolean.

Reply:
Ah my bad, I assumed it was the default, has been updated.